cms/static/js/spec/video/transcripts/videolist_spec.js

```diff
@@ -215,41 +218,32 @@ function ($, _, Utils, VideoList, MetadataModel, AbstractEditor, AjaxHelpers) {
                         }
                     ];
 
-                spyOn(view, 'getVideoObjectsList').andReturn(videoListLength);
+                spyOn(view, 'getVideoObjectsList').and.returnValue(videoListLength);
                 spyOn(view, 'openExtraVideosBar');
 
-                waitsForResponse(
-                    mockServer,
-                    function () {
-                        assertToHaveBeenRendered(videoListLength);
-                        view.getVideoObjectsList.andReturn(videoListLength);
-                        expect(view.openExtraVideosBar).toHaveBeenCalled();
-                    },
-                    function () {
-                        resetSpies(mockServer);
-                        view.render();
-                    }
-                );
-
-                waitsForResponse(
-                    mockServer,
-                    function () {
-                        assertToHaveBeenRendered(videoListHtml5mode);
-                        expect(view.openExtraVideosBar).toHaveBeenCalled();
-                    },
-                    function () {
-                        resetSpies(mockServer);
-                        view.openExtraVideosBar.reset();
-                        view.getVideoObjectsList.andReturn(videoListHtml5mode);
-                        view.render();
-                    }
-                );
+                resetSpies(this.mockServer);
+                view.render();
+
+                waitsForResponse(this.mockServer).then(function () {
+                    assertToHaveBeenRendered(videoListLength);
+                    view.getVideoObjectsList.and.returnValue(videoListLength);
+                    expect(view.openExtraVideosBar).toHaveBeenCalled();
+                });
+
+                resetSpies(this.mockServer);
```

Comment:
@ehteshamkafeel shouldn't we need to reset spies after the above `waitsForResponse` block is completed?